health_vara.py

```diff
@@ -41,10 +45,62 @@ def default_name_representation():
         # 'cjk'  => 'CJK: Family+Given'
         return 'gf'
 
+    @classmethod
+    def set_mobile(cls, parties, _field_name, value):
+        Party = Pool().get('party.party')
+        # treat whitespace values the same as None
+        value = None if (value is None) or (value == '') or (value.isspace()) else value
+
+        for party in parties:
+            mobile = next((mechanisms for
+                           mechanisms in party.contact_mechanisms
+                           if mechanisms.type == 'mobile' and mechanisms.active),
+                          None)
+
+            if value is None and mobile is None:
+                # we have nothing and want nothing, so do nothing!
+                change = None
+            elif value is None:
+                # we had a value, but want nothing, so delete it.
+                change = {'contact_mechanisms': [['delete', [mobile.id]]]}
+            elif mobile is None:
+                # we have no value, but want one, so create it.
+                change = {'contact_mechanisms': [['create', [{'type': 'mobile', 'value': value}]]]}
+            else:
+                # update the existing value with a new one.
+                change = {'contact_mechanisms': [['write', [mobile.id], {'value': value}]]}
+
+            if change:
+                Party.write([party], change)
+
+
+def get_patient_passthrough_val(patient_field_name, passthrough_config_key, default_val):
```

Comment:
Since all of these functions are acting on the patient and used directly in the class definition for the patient Model, maybe we can shorten the names to only mention the party, e.g. `get_party_value`, `get_party_field_name` and `get_party_field`. There is always a party associated with the patient and we are interested in the fields of the associated party in these functions.